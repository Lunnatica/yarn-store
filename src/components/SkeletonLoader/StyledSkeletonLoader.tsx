import styled, { keyframes } from 'styled-components';

// Keyframe for shimmer animation
const shimmerAnimation = keyframes`
  0% {
    background-position: -468px 0;
  }
  100% {
    background-position: 468px 0;
  }
`;

export const StyledSkeletonItem = styled.div`
  max-width: 298px;
  border-radius: 10px;
  height: 432px;
  background-color: #e0e0e0;

  /* Shimmer effect */
  background-image: linear-gradient(to right, #e0e0e0 0%, #f0f0f0 50%, #e0e0e0 100%);
  background-size: 800px 100%;
  animation: ${shimmerAnimation} 1.5s infinite linear;
`;
