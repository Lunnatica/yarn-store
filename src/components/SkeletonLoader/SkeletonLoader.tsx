import { StyledSkeletonItem } from './StyledSkeletonLoader';

const SKELETON_ITEMS = 6;

export const SkeletonLoader: React.FC = () => {
  return (
    <>
      {Array.from({ length: SKELETON_ITEMS }).map((_, index) => (
        <StyledSkeletonItem key={index} />
      ))}
    </>
  );
};
