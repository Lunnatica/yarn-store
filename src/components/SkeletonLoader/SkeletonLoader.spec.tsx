import { render } from '@testing-library/react';
import { SkeletonLoader } from './SkeletonLoader';

describe('SkeletonLoader', () => {
  it('should match the snapshot', () => {
    const { container } = render(<SkeletonLoader />);
    expect(container).toMatchSnapshot();
  });
});
