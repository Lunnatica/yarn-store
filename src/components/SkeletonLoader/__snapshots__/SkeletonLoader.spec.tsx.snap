// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoader should match the snapshot 1`] = `
<div>
  <div
    class="StyledSkeletonLoader__StyledSkeletonItem-sc-6m5og6-0 hFPA-dX"
  />
  <div
    class="StyledSkeletonLoader__StyledSkeletonItem-sc-6m5og6-0 hFPA-dX"
  />
  <div
    class="StyledSkeletonLoader__StyledSkeletonItem-sc-6m5og6-0 hFPA-dX"
  />
  <div
    class="StyledSkeletonLoader__StyledSkeletonItem-sc-6m5og6-0 hFPA-dX"
  />
  <div
    class="StyledSkeletonLoader__StyledSkeletonItem-sc-6m5og6-0 hFPA-dX"
  />
  <div
    class="StyledSkeletonLoader__StyledSkeletonItem-sc-6m5og6-0 hFPA-dX"
  />
</div>
`;
